### ISSUE TITLE
[1.11.2] Server crash when using descriptive book

### ISSUE THREAD

**Issue by Tori-San** [XCompWiz/Mystcraft-Issues#208]
Descriptive Books seem to crash the dedicated server. For some reason, this doesn't happen in singleplayer.

[crash report](https://github.com/XCompWiz/Mystcraft-Issues/files/1220934/crash-2017-08-13_22.35.47-server.txt)
[fml log](https://github.com/XCompWiz/Mystcraft-Issues/files/1220938/fml-server-latest.txt)
[server log](https://github.com/XCompWiz/Mystcraft-Issues/files/1220939/latest.txt)

Steps to reproduce:
* Craft a descriptive book with nothing but a linking panel page
* Put it in a bookstand and try to go through

Forge Version: `13.20.1.2386`
Mod Version: `0.13.2.00`

I made sure no other mods were installed on the server, the world and server files were just set up to reproduce this issue.

**Comment by Xirema** [comment]
Got the exact same issue, MC1.11.2, Forge 2386.

**Comment by Koyomic** [comment]
Same for me, did a bit of testing

Forge Version 1.20.1.2369
Minecraft 1.11.2

There are no other mods on the server, only Mystcraft 0.13.2.00

[Pastebin for the error](https://pastebin.com/CgSNtY6r)

I've seen a lot of people have this issue and not suggested solutions are to remove other mods, but since don't have any others, i'm stuck for what to do.

**Comment by XCompWiz** [comment]
@HellFirePvP has figured out the issue and given me a fix. Should work fin in next release. :)  Sorry about the issue!

**Comment by jpastuszek** [comment]
In the meantime is there any workaround for the issue? Mystcraft is unusable for me (dedicated server) apart from intra-demension linking or going to nether and back...
https://gist.github.com/jpastuszek/a4b47485c59e300bd8365b2a7dea0400
```
Caused by: java.lang.NoSuchMethodError: net.minecraft.world.biome.Biome.func_185361_o()I
        at com.xcompwiz.mystcraft.world.biome.BiomeWrapperMyst.generateBiomeProperties(BiomeWrapperMyst.java:45) ~[BiomeWrapperMyst.class:?]
        at com.xcompwiz.mystcraft.world.biome.BiomeWrapperMyst.<init>(BiomeWrapperMyst.java:57) ~[BiomeWrapperMyst.class:?]
        at com.xcompwiz.mystcraft.world.BiomeWrapperManager.getBiomeWrapperForBiome(BiomeWrapperManager.java:30) ~[BiomeWrapperManager.class:?]
        at com.xcompwiz.mystcraft.world.BiomeWrapperManager.getWrapper(BiomeWrapperManager.java:23) ~[BiomeWrapperManager.class:?]
        at com.xcompwiz.mystcraft.world.WorldProviderMyst.getBiomeForCoords(WorldProviderMyst.java:423) ~[WorldProviderMyst.class:?]
        at net.minecraft.world.World.func_180494_b(World.java:142) ~[ajs.class:?]
        at com.xcompwiz.mystcraft.world.ChunkProviderMyst.func_185931_b(ChunkProviderMyst.java:127) ~[ChunkProviderMyst.class:?]
        at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1001) ~[auo.class:?]
        at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:963) ~[auo.class:?]
        at net.minecraftforge.common.chunkio.ChunkIOProvider.syncCallback(ChunkIOProvider.java:105) ~[ChunkIOProvider.class:?]
        at net.minecraftforge.common.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:94) ~[ChunkIOExecutor.class:?]
        at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:119) ~[lv.class:?]
        at net.minecraft.world.gen.ChunkProviderServer.func_186028_c(ChunkProviderServer.java:90) ~[lv.class:?]
        at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:136) ~[lv.class:?]
        at net.minecraft.world.World.func_72964_e(World.java:303) ~[ajs.class:?]
        at net.minecraft.world.World.func_175726_f(World.java:298) ~[ajs.class:?]
        at net.minecraft.world.World.func_175672_r(World.java:1609) ~[ajs.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.func_76566_a(WorldProviderMyst.java:414) ~[WorldProviderMyst.class:?]
```

**Comment by scooby1350** [comment]
I have tried all the version of mystcraft and I am still getting this error. Was there a fix or work around to get mystcraft to work? I can get it to work in the stand a lone version but on the multi-player it crashes the server. Let me know please. 

[18:15:02] [Server thread/FATAL]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: net.minecraft.world.biome.Biome.func_185361_o()I
at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_91]
at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_91]
at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:679) [MinecraftServer.class:?]
at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:384) [lh.class:?]
at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:624) [MinecraftServer.class:?]
at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:482) [MinecraftServer.class:?]
at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: java.lang.NoSuchMethodError: net.minecraft.world.biome.Biome.func_185361_o()I
at com.xcompwiz.mystcraft.world.biome.BiomeWrapperMyst.generateBiomeProperties(BiomeWrapperMyst.java:45) ~[BiomeWrapperMyst.class:?]
at com.xcompwiz.mystcraft.world.biome.BiomeWrapperMyst.<init>(BiomeWrapperMyst.java:57) ~[BiomeWrapperMyst.class:?]
at com.xcompwiz.mystcraft.world.BiomeWrapperManager.getBiomeWrapperForBiome(BiomeWrapperManager.java:30) ~[BiomeWrapperManager.class:?]
at com.xcompwiz.mystcraft.world.BiomeWrapperManager.getWrapper(BiomeWrapperManager.java:23) ~[BiomeWrapperManager.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.getBiomeForCoords(WorldProviderMyst.java:423) ~[WorldProviderMyst.class:?]
at net.minecraft.world.World.func_180494_b(World.java:142) ~[ajs.class:?]
at com.xcompwiz.mystcraft.world.ChunkProviderMyst.func_185931_b(ChunkProviderMyst.java:127) ~[ChunkProviderMyst.class:?]
at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1001) ~[auo.class:?]
at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:973) ~[auo.class:?]
at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:158) ~[lv.class:?]
at net.minecraft.world.World.func_72964_e(World.java:303) ~[ajs.class:?]
at net.minecraft.world.World.func_175726_f(World.java:298) ~[ajs.class:?]
at net.minecraft.world.World.func_175672_r(World.java:1609) ~[ajs.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.func_76566_a(WorldProviderMyst.java:414) ~[WorldProviderMyst.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.verifySpawn(WorldProviderMyst.java:394) ~[WorldProviderMyst.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.getSpawnPoint(WorldProviderMyst.java:366) ~[WorldProviderMyst.class:?]
at net.minecraft.world.World.func_175694_M(World.java:3435) ~[ajs.class:?]
at com.xcompwiz.mystcraft.linking.LinkController.travelEntity(LinkController.java:49) ~[LinkController.class:?]
at com.xcompwiz.mystcraft.item.ItemLinking.activate(ItemLinking.java:86) ~[ItemLinking.class:?]
at com.xcompwiz.mystcraft.item.ItemAgebook.activate(ItemAgebook.java:124) ~[ItemAgebook.class:?]
at com.xcompwiz.mystcraft.tileentity.TileEntityBookRotateable.link(TileEntityBookRotateable.java:53) ~[TileEntityBookRotateable.class:?]
at com.xcompwiz.mystcraft.inventory.ContainerBook.processMessage(ContainerBook.java:298) ~[ContainerBook.class:?]
at com.xcompwiz.mystcraft.network.packet.MPacketGuiMessage.lambda$onMessage$0(MPacketGuiMessage.java:71) ~[MPacketGuiMessage.class:?]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_91]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_91]
at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
... 5 more
[18:15:02] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.util.ReportedException: Exception ticking world
at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:715) ~[MinecraftServer.class:?]
at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:384) ~[lh.class:?]
at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:624) ~[MinecraftServer.class:?]
at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:482) [MinecraftServer.class:?]
at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: java.lang.NoSuchMethodError: net.minecraft.world.biome.Biome.func_185361_o()I
at com.xcompwiz.mystcraft.world.biome.BiomeWrapperMyst.generateBiomeProperties(BiomeWrapperMyst.java:45) ~[BiomeWrapperMyst.class:?]
at com.xcompwiz.mystcraft.world.biome.BiomeWrapperMyst.<init>(BiomeWrapperMyst.java:57) ~[BiomeWrapperMyst.class:?]
at com.xcompwiz.mystcraft.world.BiomeWrapperManager.getBiomeWrapperForBiome(BiomeWrapperManager.java:30) ~[BiomeWrapperManager.class:?]
at com.xcompwiz.mystcraft.world.BiomeWrapperManager.getWrapper(BiomeWrapperManager.java:23) ~[BiomeWrapperManager.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.getBiomeForCoords(WorldProviderMyst.java:423) ~[WorldProviderMyst.class:?]
at net.minecraft.world.World.func_180494_b(World.java:142) ~[ajs.class:?]
at com.xcompwiz.mystcraft.world.ChunkProviderMyst.func_185931_b(ChunkProviderMyst.java:127) ~[ChunkProviderMyst.class:?]
at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1001) ~[auo.class:?]
at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:968) ~[auo.class:?]
at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:158) ~[lv.class:?]
at net.minecraft.world.World.func_72964_e(World.java:303) ~[ajs.class:?]
at net.minecraft.world.World.func_175726_f(World.java:298) ~[ajs.class:?]
at net.minecraft.world.World.func_175672_r(World.java:1609) ~[ajs.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.func_76566_a(WorldProviderMyst.java:414) ~[WorldProviderMyst.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.verifySpawn(WorldProviderMyst.java:394) ~[WorldProviderMyst.class:?]
at com.xcompwiz.mystcraft.world.WorldProviderMyst.getSpawnPoint(WorldProviderMyst.java:366) ~[WorldProviderMyst.class:?]
at net.minecraft.world.World.func_175694_M(World.java:3435) ~[ajs.class:?]
at net.minecraft.world.WorldEntitySpawner.func_77192_a(WorldEntitySpawner.java:76) ~[ajz.class:?]
at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:197) ~[lw.class:?]
at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:709) ~[MinecraftServer.class:?]
... 4 more
[18:15:02] [Server thread/ERROR]: This crash report has been saved to: /var/games/minecraft/servers/Minecraft_1_11_2/./crash-reports/crash-2017-12-06_18.15.02-server.txt

**Comment by DuncanIdaho006** [comment]
I've developed a workaround but it's clunky and won't really work for servers where you intend to play with more than a few friends.

Download the game server World file and then I load the game in single player. 
Go into the new Myst age (be sure to take a return book with you. The game may or may not crash at this point. Then close and reopen Minecraft and load into the game. The character will be in the Myst age.
Then upload the game file back to the server and start the server. 

It's a bit of a to do but it can be done and if there's only a few of us on the server it won't be that bad.
This only has to be done the first time a new world is visited.

I wouldn't recommend this on a server with more than 4 people who are actively writing ages and only if you're really good friends.

**Comment by XCompWiz** [comment]
Fix finally pushed in 0.13.2.01. Sorry for the ridiculous delay.
